src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -473,14 +477,40 @@ const vector<SynapseIdx> SpatialPooler::compute(const SDR &input, const bool lea
     bumpUpWeakColumns_();
     updateBoostFactors_();
     if (isUpdateRound_()) {
+      const UInt old_inhibitionRadius_ = inhibitionRadius_;
       updateInhibitionRadius_();
+      if (inhibitionRadius_ != old_inhibitionRadius_) { 
+	      calculateWrapAroundNeighbors();
+	      mapAllNeighbors();
+      }
       updateMinDutyCycles_();
     }
   }
 
   return overlaps;
 }
 
+void SpatialPooler::calculateWrapAroundNeighbors() {
+      wrapAroundNeighbors_ = 1;
+      for (UInt i = 0; i<columnDimensions_.size();i++) {
+	   UInt diam = 2*inhibitionRadius_ + 1;
+	   wrapAroundNeighbors_ *= std::min(diam, columnDimensions_[i]);
+      }
+      wrapAroundNeighbors_ -= 1;
+      return;
+}
+
+void SpatialPooler::mapAllNeighbors() {
+	neighborMap_.clear();
+	neighborMap_.reserve(wrapAroundNeighbors_ * numColumns_);
```

Comment:
nit, try making this an unordered_map, it'll be none (or little) slower and easier to read.

Reply:
Ok. I'll have to read up on that, I'm unfamiliar with the concept.

Reply:
Ok, should be easy enough.

Reply:
done in 
f873068